sdk/scheduler/src/main/java/com/mesosphere/sdk/debug/PlansTracker.java

```diff
@@ -0,0 +1,297 @@
+package com.mesosphere.sdk.debug;
+
+import com.mesosphere.sdk.http.ResponseUtils;
+import com.mesosphere.sdk.offer.Constants;
+import com.mesosphere.sdk.scheduler.plan.Phase;
+import com.mesosphere.sdk.scheduler.plan.Plan;
+import com.mesosphere.sdk.scheduler.plan.PlanCoordinator;
+import com.mesosphere.sdk.scheduler.plan.PlanManager;
+import com.mesosphere.sdk.scheduler.plan.Step;
+import com.mesosphere.sdk.state.StateStore;
+import com.mesosphere.sdk.state.StateStoreUtils;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+// SUPPRESS CHECKSTYLE MultipleStringLiteralsCheck
+
+/**
+ * PlansTracker is the backend of the PlansDebugEndpoint.
+ * It aggregates the status across all Plans and allows
+ * for filtered responses. The PlansTracker also returns
+ * the currently active plans and the current state the
+ * the scheduler is in.
+ */
+public class PlansTracker implements DebugEndpoint {
+
+  private final StateStore stateStore;
+
+  private final PlanCoordinator planCoordinator;
+
+  public PlansTracker(PlanCoordinator planCoordinator, StateStore stateStore) {
+    this.planCoordinator = planCoordinator;
+    this.stateStore = stateStore;
+  }
+
+  private JSONObject generateServiceStatus(String filterPlan,
+                                           String filterPhase,
+                                           String filterStep,
+                                           boolean allowFiltering,
+                                           boolean skipDetails)
+  {
+    JSONObject outcome = new JSONObject();
+    JSONArray plansArray = new JSONArray();
+
+    for (PlanManager planManager : planCoordinator.getPlanManagers()) {
+      Plan plan = planManager.getPlan();
+
+      //Filter down to a plan if specified.
+      if (allowFiltering && filterPlan != null && !plan.getName().equalsIgnoreCase(filterPlan)) {
+        continue;
+      }
+
+      JSONObject planObject = new JSONObject();
+      JSONArray phaseArray = new JSONArray();
```

Comment:
IMO... I would rather create a POJO of type we are interested in... and have a separate step what transforms to JSON..  there should be libraries that given nested POJOs will generate the output we are looking for.   This seems to conflate the gather of the info with the format we need as an output.

Reply:
perhaps this is that conversation class... I'm new to this code base..

Reply:
Yea, this class is intended for converting between the internal representation to JSON. I'll work on creating simplified POJOs, flooding them the desired state and serializing that as JSON. Lets see if that simplifies the logic here.